### PR TITLE
Fix/840

### LINKED ISSUE #840: Switch scripts/ tabular output over to proper CSV formatting.
All scripts that output tabular data should have a --csv option that outputs proper CSV, with column headers.  For khmer 1.x series, this will need to be an optional feature (due to Semantic Versioning); we can make it default for 2.x.

This has been done for count-median in #818 already, and is in process for abundance-dist in #831.

### PR DESCRIPTION
**Pull request by anotherthomas**
fixes issue #840 
Added a csv command line option for readstats. Also added missing exception handling.

### CODE REVIEW

**Comment by anotherthomas** [comment]
- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by anotherthomas** [comment]
ready for review

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,7 @@
+2015-04-14  Thomas Fenzl  <thomas.fenzk@gmx.net>
```
Comment (mr-c) [inline ChangeLog]: Newline before list, thanks :-)

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+   * scripts/{count-overlap.py,readstats.py},tests/test_scripts.py: 
+   added a --csv option to readstats, updated documentation for count-overlap
+
```
Comment (mr-c) [inline ChangeLog]: All files modified must be listed including `khmer/_khmermodule.cc` et cetera

**Comment by anotherthomas** [comment]
I've replicated the full standard output in the CSV.
After thinking about it I'm now leaning towards at least dropping the line with the summary information, since that can easily be computed by users reading the file. Thoughts?

**Comment by mr-c** [comment]
I'm good with that. @ctb ?

**Comment by ctb** [comment]
Sure

--
Titus Brown, ctbrown@ucdavis.edu

File: khmer/_khmermodule.cc [hunk 3]
```diff
@@ -1858,7 +1858,10 @@ hashbits_count_overlap(khmer_KHashbits_Object * me, PyObject * args)
     } catch (_khmer_signal &e) {
         PyErr_SetString(PyExc_IOError, e.get_message().c_str());
         return NULL;
-    }
+    } catch (InvalidStreamHandle &e) {
+        PyErr_SetString(PyExc_IOError, e.what());
```
Comment (mr-c) [inline khmer/_khmermodule.cc]: Thanks! Test please :-)

File: scripts/readstats.py [hunk 4]
```diff
@@ -51,44 +141,37 @@ def main():
...
-        bp = 0
-        seqs = 0
+    statistics = []
 
+    for filename in args.filenames:
         try:
-            input_iter = screed.open(filename, parse_description=False)
+            bps, seqs = analyze_file(filename)
         except (IOError, OSError, EOFError) as exc:
             print >>sys.stderr, 'ERROR in opening %s:' % filename
```
Comment (mr-c) [inline scripts/readstats.py]: I can haz test?

**Comment by mr-c** [comment]
Otherwise, LGTM

**Comment by anotherthomas** [comment]
@mr-c done

**Pull request merged**